cvmfs/loader_talk.cc

```diff
@@ -136,19 +136,58 @@ int MainReload(const std::string &socket_path, const bool stop_and_go,
   }
   LogCvmfs(kLogCvmfs, kLogStdout, "done");
 
-
   // reload mode: debug (d) or non-debug (n)
-  char command = debug ? 'd' : 'n';
-  WritePipe(socket_fd, &command, 1);
-
-  command = stop_and_go ? 'S' : 'R';
-  WritePipe(socket_fd, &command, 1);
+  char commands[2];
+  commands[0] = debug ? 'd' : 'n';
+  commands[1] = stop_and_go ? 'S' : 'R';
+
+  // Loaders before version 2.11 won't recognize 'd' or 'n'. They will
+  // send "unknown command" and close the connection. We try to send
+  // both the commands and react according to what we read from the socket.
+
+  ssize_t retval;
+  do {
+    retval = send(socket_fd, commands, 2, MSG_NOSIGNAL);
+  } while ((retval <= 0) && (errno == EINTR));
+
+  if (retval <= 0) {
+    LogCvmfs(kLogCvmfs, kLogStderr, "Sending reload command failed!");
+    return 103;
+  }
 
   char buf;
-  int retval;
+  std::string first_line;
+  bool past_first_line = false;
   while ((retval = read(socket_fd, &buf, 1)) == 1) {
     if (buf == '~')
       break;
+
+    if (first_line == "unknown command") {
```

Comment:
`std::string::compare`?

Reply:
`==` is more readable, compare only makes sense if you are interested in the lexical ordering

Reply:
ah thats my c/java heritage